.github/workflows/ci-cd.yml

```
@@ -43,19 +51,135 @@ jobs:
+      - name: Download build artifacts
+        uses: actions/download-artifact@v4
+        with:
+          name: build-files
+          path: dist
+          
+      - name: Set up SSH
+        uses: webfactory/ssh-agent@v0.8.0
+        with:
+          ssh-private-key: ${{ secrets.DEPLOY_SSH_KEY }}
+        # Only run this step if DEPLOY_SSH_KEY is set
+        if: ${{ secrets.DEPLOY_SSH_KEY != '' }}
```

actionlint (1.7.7)

context "secrets" is not allowed here. available contexts are "env", "github", "inputs", "job", "matrix", "needs", "runner", "steps", "strategy", "vars". see https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability for more details

(expression)